### DISCUSSION TITLE
导入到思源笔记

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#5592] [插件说明]
描述
---

将阅读模式下的内容导入到思源笔记。

地址
---

[https://simpread.ksria.cn/plugins/details/XxuF7bPTbo](https://simpread.ksria.cn/plugins/details/XxuF7bPTbo)

视频
---

https://user-images.githubusercontent.com/81074/241620554-e633513f-1379-4dcb-9629-151857b56b18.mp4

快捷键
---

- `s y` → 导入到思源笔记中

- `q s y` → 弹出快捷导入面板（可在导入前选择不同的笔记本 / 路径）

- `z s y` → 可加入到自动化辅助增强实现加入稍后读后自动的功能

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1011" alt="image" src="https://github.com/user-attachments/assets/6d3cfa2a-1923-494e-b44b-6f6b4dafaf0e">

功能
---

- 定制思源笔记地址（适用于 Docker 用户）

    <img width="560" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/d0a87350-243d-4ede-a7ef-88a21bb06337">

- 新建笔记本（留空则新建 SimpRead 名称的笔记本）

   <img width="556" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/60835a58-9052-4b35-8486-2d16612b94f1">

- 查看并选择任意笔记本

   ![2023-05-29_11-27-09 (1)](https://github.com/Kenshin/simpread/assets/81074/5249a14d-1725-4992-ac09-8412b67890ca)
   
- 自定义导入路径

   <img width="594" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/7d9f7a18-9a88-4459-a3a6-93e939a816a8">

- [思源图库](https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-6026002)

   <img width="549" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/7cda202a-ed83-44f3-86fe-cb89e6ff8362">

- [assets 定制](https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-6025979) 包括：`{{title}}` 与 `{{date}}`

- [将阅读模式的图片转换为思源图床](https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-6026002)

- 导入全文到思源笔记

- 快捷键

- [自动化](https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-6025982)

- 在导入前可选择任意笔记本或路径

   ![2023-05-29_11-36-46 (1)](https://github.com/Kenshin/simpread/assets/81074/92cd7123-4b4e-412b-bf06-7bb167a041b2)

开启图床后无法使用的解决方案
---

https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-10155344

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过[此链接](https://simpread.ksria.cn/plugins/details/XxuF7bPTbo) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

更新日志
---

### 1.1.4

- `Optimize` 优化了界面，加入了 `测试 API` 与 `测试图床 API` 功能

- `Fix`  `<sr-blockquote>` 转换错误 https://github.com/Kenshin/simpread/issues/6622

- `Fix`  当 **授权码为空** 时，无法使用思源图床的问题 https://github.com/Kenshin/simpread/issues/6311

### 1.1.3

- `Fix` MathJax 没有转换为 Katex [#5742](https://github.com/Kenshin/simpread/issues/5742)

### 1.1.2

- `Fix` 当图片无法转换为思源图床时无法正常导入的问题

### 1.1.1

- `Fix` 当没有图片或图片为 `svg` 时无法正常导入的问题

- `Add` 加入 **关于** 选项卡，包括：订阅的更新链接、群链接等

### 1.1.0

- `Add` 全新的 [选项卡](https://github.com/Kenshin/simpread/assets/81074/c18249f3-48f7-4f9a-9dc9-3393d2ac0ca0)

- `Add` 导入后显示原文链接

### 1.0.1

- `Fix` 当标题含有特殊字符时导致的错误，关联 [#5613](https://github.com/Kenshin/simpread/issues/5613)

### 1.0.0

- 定制思源笔记地址（适用于 Docker 用户）

- 新建笔记本（留空则新建 SimpRead 名称的笔记本）

- 查看并选择任意笔记本
   
- 自定义导入路径

- [思源图库](https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-6026002)

- [assets 定制](https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-6025979) 包括：`{{title}}` 与 `{{date}}`

- [将阅读模式的图片转换为思源图床](https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-6026002)

- 导入全文到思源笔记

- 快捷键

- [自动化](https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-6025982)

- 在导入前可选择任意笔记本或路径

**Comment by Kenshin** [comment]
# assets 占位符

当开启思源图床后，就可以使用此方式了。

<img width="550" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/90fd7796-2600-4bff-8e8d-222081705bde">

当此路径不填写的时候自动将图片导入到 `assets` 文件夹。

<img width="268" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/f8548c08-4b9d-4de0-b046-6bde3c0e6a40">

填写的路径均在 `assets` 文件夹下。

<img width="868" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/b5267c54-3876-44f1-9b51-2243f1c5d363">

## 占位符

同时支持两种：

| 占位符         | 描述        | 备注                                       |
|-------------|-----------|------------------------------------------|
| `{{title}}` | 当前阅读模式的标题 | 这里的标题与阅读模式的标题可能会有些区别，因为需要删除一些特殊字符才能在本地保存 |
| `{{date}}`  | 当前的日期     | e.g. `2023-05-29`                        |

### 例子

<img width="569" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/92879d48-ea95-4bc5-8553-e9b5b45e00c1">

<img width="965" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/524e6865-25fb-4348-84a4-c3c5c6d5218a">

**Comment by Kenshin** [comment]
# 自动化

除了支持手动导入到思源笔记外，还支持通过 [自动化辅助增强插件](https://github.com/Kenshin/simpread/discussions/3596) 导入到思源笔记。

将 `z s y` 加入到自动化辅助插件即可。

<img width="533" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/efd69ec4-15de-4cf8-86da-d949bc5cf1a9">

## 使用效果

https://github.com/Kenshin/simpread/assets/81074/e633513f-1379-4dcb-9629-151857b56b18

**Comment by Kenshin** [comment]
# 图床

思源笔记可以将图片以外链的形式显示 e.g.  `http://127.0.0.1:6806/assets/xxxx/1685329686937-20230529110806-ttuu9t5.png`

<img width="1130" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/a3045077-0417-4e5c-8486-d9fdef49aa47">

所以只要开着思源笔记的前提下，就相当于有了一个属于自己的图床。

当开启下图所示的选项后，可以将阅读模式的图片外链也改为思源图床的形式。

<img width="549" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/0772cbbb-33fd-4fe3-aea6-e12343a44be6">

导入思源笔记后，在阅读模式中的图片链接均自动更好为思源笔记的链接。

<img width="1636" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/8543398c-28fd-4897-b1b4-34dbb757ff4b">

如果这时，再简悦阅读模式导出 HTML 的话，导出的 HTML 中图片链接自然也是思源笔记外链形式。

### 注意

1. 如果仍需要将阅读模式的内容导入到在线的生产力工具中，则不要启用此功能。
2. 使用此功能后需要一直开着思源笔记，这样才能让外链一直正常显示。
3. 偶尔会出现无法正常转换的情况，[详细说明以及解决方案](https://github.com/Kenshin/simpread/discussions/5592#discussioncomment-6321338)。

**Reply by Kenshin** [reply]
# 为什么有些图片无法转换为思源图床

## 原因说明

个别页面会无法转换，因为触发了 **CORS 错误**，简单的解释是因为：图片的 URL 与阅读模式对应的 URL 不在同一个域下面造成的。

错误信息如下图所示

<img width="1018" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/6046ad16-f6ee-44f2-88b4-3edcb551e72b">

## 解决方案

这种情况发生的机率很低，暂时没有好的解决方案，1.1.2 版对这种方式做了优化，即便发生时也会正常导入正文到思源笔记，同时会有相应的提示。

<img width="570" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/3cd096e9-35a9-479d-8752-41d6cf2ab65f">


后续可以在思源笔记中手动转换，如下图所示。

<img width="216" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/ac79987c-22e8-4515-8b83-295177517e8a">

**Reply by childeyouyu** [reply]
请问思源图床是付费的吗？

**Reply by ljded** [reply]
> 请问思源图床是付费的吗？

并不是，思源的图床是保存在本地的，如果需要云服务才需要会员。

也可以买一个永久的基础会员然后自建云服务阿里云OSS服务40GB5年50元

**Comment by ethan1010** [comment]
如何将标注一起同步过去啊

**Reply by Kenshin** [reply]
仅支持导入全文，会考虑标注的导入，但仅限于新 Page

**Reply by ethan1010** [reply]
可以哦，非常期待

**Comment by Kenshin** [comment]
# 开启图床后无法使用的解决方案

## 问题现象

当填写 Token 并测试后，提示正确

<img width="849" alt="image" src="https://github.com/user-attachments/assets/07a3bbe4-c555-4fa4-ad84-6a8b3aa53344">

但在开启思源图床并测试后提示

<img width="849" alt="image" src="https://github.com/user-attachments/assets/96b88a27-b956-4b7d-835a-b588a391f851">

## 问题说明

右键 → 检查 → 查看控制台，会发现如下错误信息

<img width="732" alt="image" src="https://github.com/user-attachments/assets/c2c31ed3-4463-4783-a89e-3fe09e8aa351">

## 解决方案

在思源笔记中设置授权码即可

<img width="1249" alt="image" src="https://github.com/user-attachments/assets/e0ec9cf3-c391-4e12-9678-0b37b6e6d633">

<img width="497" alt="image" src="https://github.com/user-attachments/assets/4224dde5-6a5c-450b-9c73-69c7b25b09c1">

## 关联

感谢 @Zonezzc 提供的解决方案 https://github.com/Kenshin/simpread/issues/6311#issuecomment-1956766304 才能让我找到这个问题。👍 🙏

**Comment by chmdir** [comment]
没有图床怎么把图片导入到本地思源笔记中, docker自建的思源笔记

**Reply by chmdir** [reply]
simpread siyuan fail  {readyState: 0, getResponseHeader: ƒ, getAllResponseHeaders: ƒ, setRequestHeader: ƒ, overrideMimeType: ƒ, …} error 
VM148:128 simpread siyuan fail  SyntaxError: "undefined" is not valid JSON
    at JSON.parse (<anonymous>)
    at Object.eval (eval at c (contentscripts.js:1:1), <anonymous>:121:43)
    at u (common.js:7:8967)
    at Object.fireWith [as rejectWith] (common.js:7:9784)
    at n (common.js:8:14870)
    at XMLHttpRequest.<anonymous> (common.js:8:20738)
    ---
点测试图床API 失败. 思源3.1.19

**Reply by Kenshin** [reply]
什么版本？另外看下上面提到的一些解决方案，是否可以解决？

**Reply by chmdir** [reply]
简悦 思源docker 都 是最新版本
浏览器Microsoft Edge版本 131.0.2903.112 (正式版本) (64 位), 
提到的都看了,没有解决,https://github.com/Kenshin/simpread/issues/6311#issuecomment-1956766304 ,思源现在版本没有这个授权码了

**Reply by Kenshin** [reply]
你试试使用旧版本能否解决？

我目前在休假，没有办法详细测试。

**Comment by Nahuimi** [comment]
网页的表格导入到思源笔记是html代码，还要自己加工才变成表格，能不能一步到位

**Reply by Kenshin** [reply]
嗨

这是「故意」这么做的，因为 Markdown 对表格支持的不好，尤其是嵌套或比较复杂的表格无法转换为 Markdown 而一般来说现代 Markdown 编辑器都支持直接使用 HTML 表格 e.g. Obsidian、Typora 等。

1. 我会考虑优化（比如提供选项）
2. 建议思源官方支持 HTML 表格渲染方案。

**Reply by Nahuimi** [reply]
> 嗨
> 
> 这是「故意」这么做的，因为 Markdown 对表格支持的不好，尤其是嵌套或比较复杂的表格无法转换为 Markdown 而一般来说现代 Markdown 编辑器都支持直接使用 HTML 表格 e.g. Obsidian、Typora 等。
> 
> 1. 我会考虑优化（比如提供选项）
> 2. 建议思源官方支持 HTML 表格渲染方案。

思源是支持HTML 表格渲染的，用 `<div></div>`包含代码就行
![Quicker_20250223_140223](https://github.com/user-attachments/assets/733b4880-648f-492a-9f86-71706d8f6418)

**Reply by Kenshin** [reply]
> 思源是支持HTML 表格渲染的，用 `<div></div>` 包含代码就行

😂 这是非官方方案。

不过，谢谢提示，我会优化这里的。

**Comment by forealmy** [comment]
导入到思源可以加一个离线图片的选项吗。

一些做有防盗的站点，没办法导出图了。

**Reply by Kenshin** [reply]
你可以使用 图床插件实现 https://github.com/Kenshin/simpread/discussions/4306
